inc/saisegmentroute.h

```diff
@@ -149,6 +235,302 @@ typedef sai_status_t (*sai_get_segmentroute_sidlist_attribute_fn)(
+
+    /**
+     * @brief VRF for decapsulation and specific table lookup functions
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_VIRTUAL_ROUTER
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     * @validonly SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_T or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT4 or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT6 or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT46
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_VRF,
+
+    /**
+     * @brief Attach a counter
+     *
+     * When it is empty, then packet hits won't be counted
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_COUNTER
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_COUNTER_ID,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_END,
+
+    /** Custom range base value */
+    SAI_LOCAL_SID_ENTRY_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_LOCAL_SID_ENTRY_ATTR_CUSTOM_RANGE_END
+
+} sai_local_sid_entry_attr_t;
+
+/**
+ * @brief Local SID Entry
+ */
+typedef struct _sai_local_sid_entry_t
+{
+    /**
+     * @brief Switch ID
+     *
+     * @objects SAI_OBJECT_TYPE_SWITCH
+     */
+    sai_object_id_t switch_id;
+
+    /**
+     * @brief Virtual Router ID
+     *
+     * @objects SAI_OBJECT_TYPE_VIRTUAL_ROUTER
+     */
+    sai_object_id_t vr_id;
+
+    /**
+     * @brief Length of the Locator Block part of the SID
+     */
+    sai_uint8_t locator_block_len;
+
+    /**
+     * @brief Length of the Locator Node part of the SID
+     */
+    sai_uint8_t locator_node_len;
+
+    /**
+     * @brief Length of the Function part of the SID
+     */
+    sai_uint8_t function_len;
+
+    /**
+     * @brief Length of the Args part of the SID
+     */
+    sai_uint8_t args_len;
+
+    /**
+     * @brief IPv6 prefix for Local SID
+     */
+    sai_ip6_t sid;
+
+} sai_local_sid_entry_t;
+
+/**
+ * @brief Create Local SID entry
+ *
+ * @param[in] local_sid_entry Local SID entry
+ * @param[in] attr_count Number of attributes
+ * @param[in] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_create_local_sid_entry_fn)(
+        _In_ const sai_local_sid_entry_t *local_sid_entry,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list);
+
+/**
+ * @brief Remove Local SID entry
+ *
+ * @param[in] local_sid_entry Local SID entry
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_remove_local_sid_entry_fn)(
+        _In_ const sai_local_sid_entry_t *local_sid_entry);
+
+/**
+ * @brief Set Local SID attribute value
```

Comment:
Q: Is this how a sid entry modified?

Reply:
Yes, this API can be used to modify an existing SID entry.